analyzer/protocol/ldap/ldap.spicy

```diff
@@ -0,0 +1,638 @@
+                                                  &parse-from=self.filterBytes;
+    FilterType::FILTER_SUBSTR  -> FILTER_SUBSTR:  SubstringFilter(depth+1)
+                                                  &parse-from=self.filterBytes;
+    FilterType::FILTER_GE      -> FILTER_GE:      AttributeValueAssertion(depth+1)
+                                                  &parse-from=self.filterBytes;
+    FilterType::FILTER_LE      -> FILTER_LE:      AttributeValueAssertion(depth+1)
+                                                  &parse-from=self.filterBytes;
+    FilterType::FILTER_PRESENT -> FILTER_PRESENT: asn1::ASN1OctetString(self.filter.head.len.len, False)
+                                                  &convert=$$.value.decode(hilti::Charset::ASCII)
+                                                  &parse-from=self.filterBytes;
+    FilterType::FILTER_APPROX  -> FILTER_APPROX:  AttributeValueAssertion(depth+1)
+                                                  &parse-from=self.filterBytes;
+    FilterType::FILTER_EXT     -> FILTER_EXT:     asn1::ASN1Message(True, depth+1)
+                                                  &parse-from=self.filterBytes;
+  };
+};
+
+type SearchRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  baseObject: asn1::ASN1Message(True, self.depth) &convert=$$.body.octetstring;
+  scope: asn1::ASN1Message(True, self.depth) &convert=cast<SearchScope>(cast<uint8>($$.body.num_value))
+                                             &default=SearchScope::NOT_SET;
+  deref: asn1::ASN1Message(True, self.depth) &convert=cast<SearchDerefAlias>(cast<uint8>($$.body.num_value))
+                                             &default=SearchDerefAlias::NOT_SET;
+  sizeLimit: asn1::ASN1Message(True, self.depth) &convert=$$.body.num_value &default=0;
+  timeLimit: asn1::ASN1Message(True, self.depth) &convert=$$.body.num_value &default=0;
+  typesOnly: asn1::ASN1Message(True, self.depth) &convert=$$.body.bool_value &default=False;
+  filter: SearchFilter(self.depth);
+  attributes: AttributeSelection(self.depth);
+
+  on %done {
+    message.obj = self.baseObject;
+    message.arg = "%s" % self.scope;
+    print "SearchRequest", message.messageID, self;
+  }
+};
+
+type SearchResultEntry = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  objectName: asn1::ASN1Message(True, self.depth) &convert=$$.body.octetstring;
+  # TODO: if we want to descend down into PartialAttributeList
+  attributes: asn1::ASN1Message(True, self.depth);
+
+  on %done {
+    message.obj = self.objectName;
+    print "SearchResultEntry", message.messageID, self;
+  }
+};
+
+type SearchResultDone = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
```

Comment:
Let's move this into a `%done` hook for `result` and then potentially hide the field from users after consuming its data.